### PR TITLE
Correct copyright notices to reflect Copyright OpenSearch Contributors

### LINKED ISSUE #28: Correct copyright notices to reflect Copyright OpenSearch Contributors
Coming from https://github.com/opensearch-project/.github/issues/21.

The correct copyright for open-source projects in opensearch-project is "Copyright OpenSearch Contributors". Please correct any places that say otherwise, especially where it says copyright Amazon. Make sure NOTICE.txt and README match. See https://github.com/opensearch-project/.github/pull/24 for an example.

### PR DESCRIPTION
**Pull request by eirsep**
Signed-off-by: Surya Nistala <snistala@amazon.com>

### Description
Adds correct copyright notices to reflect Copyright OpenSearch Contributors
 
### Issues Resolved
#28 
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: NOTICE [hunk 1]
```diff
@@ -1,9 +1,2 @@
 OpenSearch
 Copyright 2021 OpenSearch Contributors
```
Comment (AMoo-Miki) [inline NOTICE:2]: Please refer to [this](https://github.com/opensearch-project/opensearch-plugins/blob/main/NOTICE) for NOTICE file.

File: README.md [hunk 2]
```diff
@@ -119,3 +119,7 @@ See [CONTRIBUTING](CONTRIBUTING.md#security-issue-notifications) for more inform
 ## License
 
 This project is licensed under the Apache-2.0 License.
+
+## Copyright
+
+Copyright OpenSearch Contributors. See [NOTICE](NOTICE.txt) for details.
```
Comment (AMoo-Miki) [inline README.md:125]: The correct link to the NOTICE file is `NOTICE` without the `.txt`.

**Review by AMoo-Miki — CHANGES_REQUESTED** [review]

**Comment by AMoo-Miki** [comment]
Additionally, I noticed files like [this one](https://github.com/opensearch-project/asynchronous-search/blob/main/build-tools/plugin-coverage.gradle) still have the Amazon.com copyright headers. Please refer to this document for [the correct](https://github.com/opensearch-project/opensearch-plugins/blob/main/HEADERS.md) headers.

**Review by AMoo-Miki — APPROVED** [review]
Thanks a ton. This looks awesome.

**Review by CEHENKLE — APPROVED** [review]

**Pull request merged**